`DeleteThenPoll` should also check for 404 in the response

After switching `ServiceBus` over several tests began failing with the same error

```
testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
Error: listing resources in Resource Group: (Resource Group "acctestRG-220622150106398156"): resources.Client#ListByResourceGroup: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'acctestRG-220622150106398156' could not be found."
```

This could be a bug in the deletions of the ServiceBus SDK where the DeleteOperation isn't expecting and doesn't know how to handle a 404 in the response when polling e.g.

```
namespace Pandora.Definitions.ResourceManager.ServiceBus.v2021_06_01_preview.Namespaces;

internal class DeleteOperation : Operations.DeleteOperation
{
    public override IEnumerable<HttpStatusCode> ExpectedStatusCodes() => new List<HttpStatusCode>
        {
                HttpStatusCode.Accepted,
                HttpStatusCode.NoContent,
                HttpStatusCode.OK,
        };

    public override bool LongRunning() => true;

    public override ResourceID? ResourceId() => new NamespaceId();
}
```

--- a/tools/generator-go-sdk/generator/templater_methods_autorest.go
+++ b/tools/generator-go-sdk/generator/templater_methods_autorest.go
@@ -683,7 +683,7 @@ type %[1]s struct {
 
 func (c methodsAutoRestTemplater) senderLongRunningOperationTemplate(data ServiceGeneratorData) string {
 	checkForNotFound := ""
-	if c.operationName == "Delete" {
+	if strings.EqualFold(c.operation.Method, "DELETE") {
 		checkForNotFound = fmt.Sprintf(`
 	if !response.WasNotFound(future.Poller.HttpResponse) {
     	return future, err

--- a/tools/generator-go-sdk/generator/templater_methods_autorest_test.go
+++ b/tools/generator-go-sdk/generator/templater_methods_autorest_test.go
@@ -1,6 +1,7 @@
 package generator
 
 import (
+	"github.com/hashicorp/pandora/tools/sdk/resourcemanager"
 	"testing"
 )
 
@@ -11,7 +12,12 @@ func TestTemplateMethodsAutoRestLRODelete(t *testing.T) {
 		source:            AccTestLicenceType,
 	}
 
-	actual := methodsAutoRestTemplater{operationName: "Delete"}.senderLongRunningOperationTemplate(input)
+	actual := methodsAutoRestTemplater{
+		operation: resourcemanager.ApiOperation{
+			Method: "DELETE",
+		},
+		operationName: "Delete",
+	}.senderLongRunningOperationTemplate(input)
 
 	expected := `
 // senderForDelete sends the Delete request. The method will close the
@@ -30,7 +36,7 @@ func (c pandaClient) senderForDelete(ctx context.Context, req *http.Request) (fu
 	return
 }
 `
-	assertTemplatedCodeMatches(t, actual, expected)
+	assertTemplatedCodeMatches(t, expected, actual)
 }
 
 func TestTemplateMethodsAutoRestLROCreate(t *testing.T) {
@@ -40,7 +46,12 @@ func TestTemplateMethodsAutoRestLROCreate(t *testing.T) {
 		source:            AccTestLicenceType,
 	}
 
-	actual := methodsAutoRestTemplater{operationName: "Create"}.senderLongRunningOperationTemplate(input)
+	actual := methodsAutoRestTemplater{
+		operation: resourcemanager.ApiOperation{
+			Method: "PUT",
+		},
+		operationName: "Create",
+	}.senderLongRunningOperationTemplate(input)
 
 	expected := `
 // senderForCreate sends the Create request. The method will close the
@@ -57,5 +68,5 @@ func (c pandaClient) senderForCreate(ctx context.Context, req *http.Request) (fu
 	return
 }
 `
-	assertTemplatedCodeMatches(t, actual, expected)
+	assertTemplatedCodeMatches(t, expected, actual)
 }

